### PR TITLE
Fix lastest msys2 windows build

### PR DESCRIPTION
**Pull request by bernie-laberge**
### Linked issues
NA

### Summarize your change.

- Added the required SETUPTOOLS_USE_DISTUTILS=stdlib instruction to the documentation when installing the python requirements on Windows
- Fixed the rvsetup alias so that on Windows, it contains the required SETUPTOOLS_USE_DISTUTILS=stdlib instruction.
- Added a note to indicate that XCode 14.3.1 must be used to build Open RV on Mac (as FFmpeg cannot be built with a more recent XCode version such as XCode 15 for example)
- Added a graphical icon to indicate which msys2 shell to use to build RV.
- Added the new ACLOCAL_PATH env var requirement to build RV to the windows build instructions.

### Describe the reason for the change.
A recent msys2 update made Open RV fail to build when installed.

### Describe what you have tested and on which operating system.

Successful build was validated on Windows and Mac

### If possible, provide screenshots.

### CODE REVIEW

File: docs/build_system/config_windows.md [hunk 1]
```diff
@@ -82,9 +84,12 @@ Note: To confirm which version/location of any tool used inside the MSYS shell,
 
 ### Setting the PATH
 
-The path to mingw's bin must be added e.g. `msys64/mingw64/bin`.
+The path to MSYS2's mingw64/bin folder must be added to the path.
+To set your PATH correctly: edit the MSYS2's ~/.bashrc file and add the following line:
+`PATH=/c/msys64/mingw64/bin:PATH`
```
Comment (cedrik-fuoco-adsk) [inline docs/build_system/config_windows.md:89]: Shouldn't it be $PATH?

Comment (bernie-laberge) [reply docs/build_system/config_windows.md:89]: Totally ! 
I am using this:
PATH=/c/msys64/mingw64/bin:${PATH}
I'll change the doc. Thanks !

File: rvcmds.sh [hunk 2]
```diff
@@ -47,7 +47,7 @@ RV_BUILD_PARALLELISM="${RV_BUILD_PARALLELISM:-$(python3 -c 'import os; print(os.
 
 # ALIASES: Basic commands
 
-alias rvsetup="python3 -m pip install --user --upgrade -r ${RV_HOME}/requirements.txt"
+alias rvsetup="SETUPTOOLS_USE_DISTUTILS=${SETUPTOOLS_USE_DISTUTILS} python3 -m pip install --user --upgrade -r ${RV_HOME}/requirements.txt"
 alias rvcfg="cmake -B ${RV_BUILD} -G \"${CMAKE_GENERATOR}\" ${CMAKE_WIN_ARCH} -DCMAKE_BUILD_TYPE=Release -DRV_DEPS_QT5_LOCATION=${QT_HOME} -DRV_DEPS_WIN_PERL_ROOT=${WIN_PERL}"
```
Comment (cedrik-fuoco-adsk) [inline rvcmds.sh:51]: Question; What does this change fix?

File: README.md [hunk 3]
```diff
@@ -79,6 +79,11 @@ Use the `rvsetup` common build alias to run the bootstrap step.
 python3 -m pip install --user --upgrade -r requirements.txt
 ```
 
+Note that on Windows, use the following command instead from an MSYS2-MinGW64 shell:
+```bash
+SETUPTOOLS_USE_DISTUTILS=stdlib python3 -m pip install --user --upgrade -r requirements.txt
+```
+
```
Comment (cedrik-fuoco-adsk) [inline README.md:86]: Non-blocking, but It seems like SETUPTOOLS_USE_DISTUTILS is deprecated in 3.12 (or will)?

Reference:
https://github.com/mhammond/pywin32/issues/2119
https://github.com/pypa/setuptools/issues/3625

Comment (bernie-laberge) [reply README.md:86]: Without it, some required packages cannot be installed on windows with msys2.

Comment (cedrik-fuoco-adsk) [reply README.md:86]: I was wondering as I didn't have any issues without this on my setup.

**Review by cedrik-fuoco-adsk — APPROVED** [review]

**Review by geffrak — APPROVED** [review]

File: rvcmds.sh [hunk 4]
```diff
@@ -47,7 +47,7 @@ RV_BUILD_PARALLELISM="${RV_BUILD_PARALLELISM:-$(python3 -c 'import os; print(os.
 
 # ALIASES: Basic commands
 
-alias rvsetup="python3 -m pip install --user --upgrade -r ${RV_HOME}/requirements.txt"
+alias rvsetup="SETUPTOOLS_USE_DISTUTILS=${SETUPTOOLS_USE_DISTUTILS} python3 -m pip install --user --upgrade -r ${RV_HOME}/requirements.txt"
```
Comment (rogernelson) [inline rvcmds.sh:50]: Could that cause issues with non-Windows?

Comment (bernie-laberge) [reply rvcmds.sh:50]: I tested it on Mac and it is working fine : SETUPTOOLS_USE_DISTUTILS=

**Pull request merged**